src/acl_device.cpp

```diff
@@ -676,7 +676,7 @@ CL_API_ENTRY cl_int CL_API_CALL clRetainDevice(cl_device_id device) {
 
 ACL_EXPORT CL_API_ENTRY cl_int CL_API_CALL
 clReleaseDeviceIntelFPGA(cl_device_id device) {
-  acl_lock();
+  acl_lock_guard lock{acl_wrapper_mutex};
```

Comment:
For this, I thought we discussed about this in a previous thread. std::scoped_lock has the try_lock() requirement for the mutex. However, our underlying implementation doesn't allow us to implement try_lock(). To use it will be a violation of the requirements of std::scoped_lock(). Therefore, to do locking with scoped_lock() may also lead to issues in this case which introduces risks and we need to do more testing. Again, I don't have anymore bandwidth for this...